python/cudf/cudf/pandas/fast_slow_proxy.py

```diff
@@ -1053,7 +1073,9 @@ def _fast_slow_function_call(
             domain="cudf_pandas",
         ):
             slow_args, slow_kwargs = _slow_arg(args), _slow_arg(kwargs)
-            if _env_get_bool("CUDF_PANDAS_FAIL_ON_FALLBACK", False):
+            if _env_get_bool("CUDF_PANDAS_WARN_ON_FALLBACK", False):
+                _warn_fallback_occurred(err, slow_args[0].__qualname__)
+            elif _env_get_bool("CUDF_PANDAS_FAIL_ON_FALLBACK", False):
```

Comment:
Since it's mostly for developer testing and we know not to do it it's probably fine, but we should be aware that with this logic warn on fallback explicitly supersedes fail on fallback in this if/else cascade.